modules/grub.sh

```diff
@@ -31,11 +31,14 @@ EOF
 
     # For partuuid support grub.cfg expects dedicated variables to be added
     if [ "${MENDER_ENABLE_PARTUUID}" == "y" ]; then
+        boot_partuuid=$(disk_get_partuuid_from_device "${boot_part_device}")
         rootfsa_partuuid=$(disk_get_partuuid_from_device "${root_part_a_device}")
         rootfsb_partuuid=$(disk_get_partuuid_from_device "${root_part_b_device}")
+        log_info "Using boot partition partuuid in grubenv: $boot_partuuid"
         log_info "Using root partition A partuuid in grubenv: $rootfsa_partuuid"
         log_info "Using root partition B partuuid in grubenv: $rootfsb_partuuid"
         cat <<- EOF >> work/grub-mender-grubenv-${MENDER_GRUBENV_VERSION}/mender_grubenv_defines
+mender_boot_uuid=${boot_partuuid}
 mender_rootfsa_uuid=${rootfsa_partuuid}
 mender_rootfsb_uuid=${rootfsb_partuuid}
 EOF
```

Comment:
For coherence, I would now move the two inconditional lines above:
```
mender_rootfsa_part=${MENDER_ROOTFS_PART_A_NUMBER}
mender_rootfsb_part=${MENDER_ROOTFS_PART_B_NUMBER}
```
to inside the `else` that follows, something like:
```
    else
        cat <<- EOF >> work/grub-mender-grubenv-${MENDER_GRUBENV_VERSION}/mender_grubenv_defines
mender_kernel_root_base=${MENDER_STORAGE_DEVICE_BASE}
mender_rootfsa_part=${MENDER_ROOTFS_PART_A_NUMBER}
mender_rootfsb_part=${MENDER_ROOTFS_PART_B_NUMBER}
EOF
```

Reply:
The `mender_rootfsa_part` and `mender_rootfsb_part` are still needed. This is somewhat historical and definitely not optimal, but we still rely on simple one or two digit partition indexes when interacting with the bootloader from the Mender client. For example, if `mender_rootfsa_part` and `mender_rootfsb_part` are set to 2 and 3, then we know that those indexes refer to `mender_rootfsa_uuid` and `mender_rootfsb_uuid`, respectively. The key variable, `mender_boot_part` is only set to the indexes, not the partition UUIDs, so we need this mapping at boot time to know which partition UUID to select.